src/tests/detect-mqtt.c

```diff
@@ -0,0 +1,126 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Sascha Steinbiss <sascha@steinbiss.name>
+ */
+
+#include "suricata-common.h"
+#include "util-unittest.h"
+
+#include "rust.h"
+
+/**
+ * \test MQTTProtocolVersionTestParse01 is a test for a valid value
+ *
+ *  \retval 1 on success
+ *  \retval 0 on failure
+ */
+static int MQTTProtocolVersionTestParse01(void)
```

Comment:
these could all be SV tests, it seems

Reply:
Why ?

This is just signatures parsing.
It could rather be unit tests of the parsing function, which is just the generic integer one already tested...

Reply:
Also, the keyword is already tested in SV with mqtt-pub-rules and mqtt-unsub-rules

Reply:
the tests only test that the rules parse w/o error, nothing internal is checked. So it's not really unittesting.

Reply:
I think the intent was the parsing of the keyword parameters... Proposing something in next version